packages/mc-scripts/static.js

```diff
@@ -0,0 +1,125 @@
+}
+
+const useLocalAssets = flags['use-local-assets'];
+
+const appDirectory = fs.realpathSync(process.cwd());
+const resolveApp = relativePath => path.resolve(appDirectory, relativePath);
+
+const paths = {
+  envPath: flags.config,
+  cspPath: flags.csp,
+  publicAssetsPath: resolveApp('public'),
+  // NOTE: previously, for running the prod server locally, we were copying
+  // assets into public/assets and compiling the index.html into public folder.
+  // To run the app then we would define the cdnUrl as http://localhost:3001/assets,
+  // because of the assets subfolder.
+  // However, on the webpack-dev-server, the index.html and the bundles are
+  // served from the same folder. It's really complicated to get the config right
+  // and eventually it's not worth the effort as it just feels like a bloody workaround.
+  // Therefore, we do it a bit different now: keep the webpack-dev-server
+  // config as is and adjust the mc-script to serve the files the same way.
+  // Remember that for normal production usage, assets are served from a CDN.
+  // So now we have a flat public folder instead of an assets subfolder.
+  // Which means that the 3 static files (favicon, google**.html, robots.txt)
+  // need to be put somewhere else (public-assets) and copy into the public
+  // folder when the server starts.
+  publicAssetsFolderPath: path.join(
+    `${applicationStaticAssetsPath}/html-page/*`
+  ),
+};
+
+shelljs.rm('-rf', paths.publicAssetsPath);
+shelljs.mkdir('-p', paths.publicAssetsPath);
+shelljs.cp('-R', paths.publicAssetsFolderPath, paths.publicAssetsPath);
+
+// This should only be used locally, as we're relying on relative paths
+// outside of this package.
+if (useLocalAssets) {
+  // Resolve the absolute path of the caller location. This is necessary
+  // to point to files within that folder.
+  const assetsFrom = resolveApp('dist/assets');
+  // Make sure that the `dist/assets` folder is available.
+  try {
+    fs.accessSync(assetsFrom, fs.F_OK);
+  } catch (error) {
+    throw new Error(
+      'Could not find "dist/assets" folder. Did you run `yarn build` before starting the server?'
+    );
+  }
+  // Copy the `dist/assets` folder into the `public` folder.
+  shelljs.cp('-R', path.join(assetsFrom, '/*'), paths.publicAssetsPath);
+}
+
+const generateStatic = async () => {
+  const compiled = await compileHtml({
+    envPath: paths.envPath,
+    cspPath: paths.cspPath,
+    publicAssetsPath: paths.publicAssetsPath,
+    useLocalAssets,
+  });
+
+  fs.writeFileSync(
+    path.join(paths.publicAssetsPath, 'index.html'),
+    compiled.indexHtmlContent,
+    { encoding: 'utf8' }
+  );
+
+  if (flags.transformer) {
+    try {
+      require.resolve(flags.transformer);
+    } catch (error) {
+      throw new Error(
+        `Could not load transformer module "${flags.transformer}"\n${
+          error.stack
+        }`
+      );
+    }
+    const transformerFn = require(flags.transformer);
+    transformerFn(compiled);
```

Comment:
This is the where we pass the compiled data to the transformer module

Reply:
What does `yarn static` do? I'm confused about the naming. Does it build static version of the sites? Wouldn't it make more sense to include build in the name then? 😕

Reply:
I couldn't think of a better name. Do you suggest something like `build-static`?

What it does is basically two things:
- it compiles the `index.html.template` into `index.html`
- it moves all the assets (including some extra ones) into a `public` folder